common/cli.h

```diff
@@ -0,0 +1,259 @@
+    //     cmd.add( arg1 );
+public:
+    template< typename... Rest >
+    cli_no_context & arg( TCLAP::Arg & a )
+    {
+        super::add( a );
+        return *this;
+    }
+    template< typename... Rest >
+    cli_no_context & arg( TCLAP::Arg & a, Rest... rest )
+    {
+        super::add( a );
+        return arg( std::forward< Rest >( rest )... );
+    }
+
+    cli_no_context & default_log_level( rs2_log_severity severity )
+    {
+        _default_log_level = severity;
+        return *this;
+    }
+
+    // Replace parse() with our own process()
+protected:
+    using super::parse;  // don't let user call parse() - enforce using process()
+
+    virtual void apply()
+    {
+#ifdef BUILD_EASYLOGGINGPP
+        if( debug_arg.getValue() )
+            rsutils::os::ensure_console();
+        if( rsutils::os::has_console() )
+            rs2_log_to_console( debug_arg.getValue() ? RS2_LOG_SEVERITY_DEBUG : _default_log_level, nullptr );
+#endif
+    }
+
+    virtual rsutils::json build_cli_settings()
+    {
+        return rsutils::json::object();
+    }
+
+public:
+    rsutils::json process( int argc, const char * const * argv )
+    {
+        parse( argc, argv );
+        apply();
+        return build_cli_settings();
+    }
+
+protected:
+    // Override basic output methods:
+    // If help is needed, for example, usage() will get called but we have not even ensured we have a console yet!
+    class cmdline_output : public TCLAP::StdOutput
+    {
+        using super = TCLAP::StdOutput;
+
+    public:
+        void usage( TCLAP::CmdLineInterface & c ) override
+        {
+            rsutils::os::ensure_console();
+            super::usage( c );
+        }
+
+        void version( TCLAP::CmdLineInterface & c ) override
+        {
+            rsutils::os::ensure_console();
+            super::version( c );
+        }
+
+        void failure( TCLAP::CmdLineInterface & c, TCLAP::ArgException & e ) override
+        {
+            rsutils::os::ensure_console();
+            super::failure( c, e );
+        }
+    };
+
+    cmdline_output _our_cmdline_output;
+
+    // Public args that can be retrieved by the user
+public:
+    flag debug_arg;
+};
+
+
+// CLI with additional context-controlling behavior:
+//      (none at this time)
+//
+class cli_no_dds : public cli_no_context
+{
+    using super = cli_no_context;
+
+public:
+    using super::super;  // inherit same ctors
+};
+
+
+// CLI with additional DDS-controlling behavior:
+//      --eth
+//      --eth-only
+//      --no-eth
+//
+class cli : public cli_no_dds
+{
+    using super = cli_no_dds;
+
+public:
+    cli( std::string const & tool_name, std::string const & version_string )
+        : super( tool_name, version_string )
+        , eth_arg( "eth", "Use DDS devices, even if disabled" )
+        , eth_only_arg( "eth-only", "Use ONLY DDS devices; do not look for USB/MIPI devices" )
+        , no_eth_arg( "no-eth", "Disable DDS devices, even if enabled" )
```

Comment:
`eth/no_eth_arg` description not that clear. Maybe something like "Don't detect DDS devices, even if DDS is enabled in config file"

Reply:
Now "Do not detect DDS devices, even if enabled in the configuration" -- good?

Reply:
Yes. Also update `eth_arg` description.